controllers/front/OrderDetailController.php

```diff
@@ -108,17 +108,31 @@ public function postProcess()
                     }
 
                     if (Validate::isLoadedObject($customer)) {
-                        Mail::Send($this->context->language->id, 'order_customer_comment', Mail::l('Message from a customer'),
-                        array(
-                            '{lastname}' => $customer->lastname,
-                            '{firstname}' => $customer->firstname,
-                            '{email}' => $customer->email,
-                            '{id_order}' => (int)$order->id,
-                            '{order_name}' => $order->getUniqReference(),
-                            '{message}' => Tools::nl2br($msgText),
-                            '{product_name}' => $product_name
-                        ),
-                        $to, $toName, $customer->email, $customer->firstname.' '.$customer->lastname);
+                        Mail::Send(
+                            $this->context->language->id,
+                            'order_customer_comment',
+                            Mail::l('Message from a customer'),
+                            array(
+                                '{lastname}' => $customer->lastname,
+                                '{firstname}' => $customer->firstname,
+                                '{email}' => $customer->email,
+                                '{id_order}' => (int)$order->id,
+                                '{order_name}' => $order->getUniqReference(),
+                                '{message}' => Tools::nl2br($msgText),
+                                '{product_name}' => $product_name
+                            ),
+                            $to,
+                            $toName,
+                            strval(Configuration::get('PS_SHOP_EMAIL')),
```

Comment:
Hi @mcdado ,

Maybe I can help by rephrasing your solution :)

Let's say a Mary Dinkelplotz (mary.dinkelplotz@gmail.com) leaves a comment with her order during a purchase on shopdomain.com. 
Then the email received by the shop's owner has the following properties :
MAIL FROM: main.address@shopdomain.com
Sender: mary.dinkelplotz@gmail.com (<- here I'm not sure it's OK)
From: "Mary Dinkelplotz" <main.address@shopdomain.com>
To: main.address@shopdomain.com
Reply to:  "Mary Dinkelplotz" <mary.dinkelplotz@gmail.com>

Is that right ? ^^